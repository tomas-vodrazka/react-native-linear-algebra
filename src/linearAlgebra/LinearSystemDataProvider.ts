import * as mathjs from "mathjs";
import { LinearSystem } from "./types";

function createMathjsContext(seed?: string) {
  if (!seed) {
    return mathjs;
  }

  return mathjs.create(mathjs.all, {
    randomSeed: seed
  });
}

export function getRandomLinearSystem(
  rows: number,
  cols: number,
  seed?: string
): LinearSystem {
  const math = createMathjsContext(seed);
  return {
    x: math.random([rows, cols]) as number[][],
    y: math.random([rows]) as number[]
  };
}
