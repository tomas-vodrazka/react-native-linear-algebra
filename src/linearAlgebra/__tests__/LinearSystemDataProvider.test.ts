import { getRandomLinearSystem } from "../LinearSystemDataProvider";

describe("LinearSystemDataProvider", () => {
  describe("getRandomLinearSystem", () => {
    it("returns a linear system", () => {
      const rows = 10;
      const cols = 5;
      const actual = getRandomLinearSystem(rows, cols);

      expect(actual.x.length).toBe(rows);
      expect(actual.x[0].length).toBe(cols);
      expect(actual.y.length).toBe(rows);
    });

    it("returns a linear system for a random seed", () => {
      const rows = 10;
      const cols = 5;
      const actual = getRandomLinearSystem(rows, cols, "random_seed");

      expect(actual).toMatchSnapshot();
    });
  });
});
