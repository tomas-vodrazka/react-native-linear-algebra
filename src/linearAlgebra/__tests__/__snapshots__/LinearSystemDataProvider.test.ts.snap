// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinearSystemDataProvider getRandomLinearSystem returns a linear system for a random seed 1`] = `
Object {
  "x": Array [
    Array [
      0.598581516914215,
      0.06231463079092458,
      0.15679548572190793,
      0.08037555910359896,
      0.21467344790675078,
    ],
    Array [
      0.5365518848631233,
      0.37597233882976105,
      0.118456582013232,
      0.4029255165578738,
      0.21797388633646886,
    ],
    Array [
      0.87910129300011,
      0.6026716626839745,
      0.8229856622460349,
      0.45334000110177824,
      0.3846403929839695,
    ],
    Array [
      0.5965731761112294,
      0.20682527003163526,
      0.23913093316194745,
      0.6669576973824195,
      0.027149158435634556,
    ],
    Array [
      0.7033184406038695,
      0.4841324089871702,
      0.8279829137417655,
      0.7615066526847286,
      0.23897337249030434,
    ],
    Array [
      0.18480924586911696,
      0.17284629418474154,
      0.8042694038059938,
      0.9011355136411101,
      0.7412912638007285,
    ],
    Array [
      0.5418808999398024,
      0.889684410446954,
      0.9141693209634829,
      0.6388273280979327,
      0.9263718030978756,
    ],
    Array [
      0.8400895530797226,
      0.4444755505072167,
      0.18488294841659497,
      0.16463904782936845,
      0.7833918927421394,
    ],
    Array [
      0.599450017431635,
      0.19521979322358396,
      0.8172327503517427,
      0.6609128840818407,
      0.4962223255301441,
    ],
    Array [
      0.528245309357787,
      0.4835758993462867,
      0.575314782574048,
      0.9517511291266735,
      0.9379011250066557,
    ],
  ],
  "y": Array [
    0.5615253913012431,
    0.758400830732582,
    0.6647419566313348,
    0.022718969466561587,
    0.003786779637359863,
    0.2607359151609309,
    0.7331887324365352,
    0.045303388461064076,
    0.2485241001555018,
    0.11893057362597625,
  ],
}
`;
